### PR TITLE
Render separate projects for argocd and for future fleet

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/16862

1. management-cluster-argocd with two applications:
  1.1 management-clusters-fleet-argocd, which syncs build/argocd/install.yaml (same for all providers)
  1.2 management-clusters-fleet-collections, which syncs this project configuration
2. management-cluster-fleet project for collection application

### CODE REVIEW

File: manifests/base/mc-fleet-apps.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    namespace: argocd
+  syncPolicy:
+    automated:
+      prune: true 
+      allowEmpty: true
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: management-clusters-fleet-app-provider
```
Comment (kopiczko) [inline manifests/base/mc-fleet-apps.yaml:24]:
```suggestion
  name: management-clusters-fleet-provider
```

File: manifests/provider/azure/override.yaml [hunk 2]
```diff
@@ -12,10 +12,18 @@ data:
...
-  description: Management Cluster Azure fleet
+  description: Management Cluster Azure collections
   sourceRepos:
     - 'https://github.com/giantswarm/azure-app-collection'
-
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: management-clusters-fleet-app-provider
```
Comment (kopiczko) [inline manifests/provider/azure/override.yaml:25]:
```suggestion
  name: management-clusters-fleet-provider
```

File: manifests/provider/kvm/override.yaml [hunk 3]
```diff
@@ -12,10 +12,18 @@ data:
...
-  description: Management Cluster KVM fleet
+  description: Management Clusters KVM Collections
   sourceRepos:
     - 'https://github.com/giantswarm/kvm-app-collection'
-
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: management-clusters-fleet-app-provider
```
Comment (kopiczko) [inline manifests/provider/kvm/override.yaml:25]:
```suggestion
  name: management-clusters-fleet-provider
```

File: manifests/provider/vmware/override.yaml [hunk 4]
```diff
@@ -12,10 +12,18 @@ data:
...
-  description: Management Cluster VMWare fleet
+  description: Management Cluster VMWare collections
   sourceRepos:
     - 'https://github.com/giantswarm/vmware-app-collection'
-
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: management-clusters-fleet-app-provider
```
Comment (kopiczko) [inline manifests/provider/vmware/override.yaml:25]:
```suggestion
  name: management-clusters-fleet-provider
```

File: manifests/base/mc-collections-project.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  sourceRepos:
+    - '*'
+  destinations:
+    - namespace: giantswarm
+      server: https://kubernetes.default.svc
+    - namespace: monitoring
+      server: https://kubernetes.default.svc
+    - namespace: kube-system
+      server: https://kubernetes.default.svc
+    - server: https://kubernetes.default.svc
```
Comment (kopiczko) [inline manifests/base/mc-collections-project.yaml:19]: I'd put the project and apps in the same file as we do for the overwrites.

File: manifests/base/mc-fleet-apps.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (kopiczko) [inline manifests/base/mc-fleet-apps.yaml:1]: I'd expand the file name to `management-clusters-fleet.yaml`

File: manifests/base/mc-fleet-apps.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: management-clusters-fleet-app
+  namespace: argocd
+spec:
+  project: management-clusters-fleet
```
Comment (kopiczko) [inline manifests/base/mc-fleet-apps.yaml:8]: Maybe this makes sense to make that more obvious and aligned with the build directory layout?

```suggestion
  project: management-clusters-fleet-argocd
```

File: manifests/base/mc-collections-project.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: management-clusters-collections
```
Comment (kopiczko) [inline manifests/base/mc-collections-project.yaml:5]: Only a suggestion:

```suggestion
  name: management-clusters-fleet-collections
```

**Review by kopiczko — COMMENTED** [review]
In general this LGTM, I'm leaving some suggestions but none of them are critical to me.

**Comment by corest** [comment]
Addressed comments

File: README.md [hunk 9]
```diff
@@ -10,5 +10,8 @@ Run
 hack/update-argocd-manifests.sh <version>
 ```
 
-This script will render `manifests/install.yaml` file with all the required resources.
+This script will render:
+  - `build/argocd/install.yaml` file with all the upstream resources
+  - `build/mc-fleet/<provider>.yaml` file with the project for self-managed provider application.
```
Comment (kopiczko) [inline README.md:15]:
```suggestion
  - `build/provider/<provider>.yaml` file with the project for self-managed provider application.
```

File: hack/update-argocd-manifests.sh [hunk 10]
```diff
@@ -28,8 +28,15 @@ cd -
 
 sed -i "/dex/d" manifests/base/kustomization.yaml
 
-echo "${AUTOGENMSG}" > "${SRCROOT}/build/install.yaml"
-${KUSTOMIZE} build "manifests/cluster-install" >> "${SRCROOT}/build/install.yaml"
+mkdir -p ${SRCROOT}/build/argocd
+echo "${AUTOGENMSG}" > "${SRCROOT}/build/argocd/install.yaml"
+${KUSTOMIZE} build "manifests/cluster-install" >> "${SRCROOT}/build/argocd/install.yaml"
```
Comment (kopiczko) [inline hack/update-argocd-manifests.sh:33]: `cluster-install`?

**Review by kopiczko — COMMENTED** [review]
Two comments, I'm trying to reproduce this again in my head

File: hack/update-argocd-manifests.sh (hunk 10 above)
Comment (corest) [reply hack/update-argocd-manifests.sh:33]: This comes from upstream argocd repo

Comment (kopiczko) [reply hack/update-argocd-manifests.sh:33]: Okay I know what that is.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**